block-library/tabs/blocks/tabs/edit.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * WordPress Dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+import {
+	useBlockProps,
+	RichText,
+	useInnerBlocksProps,
+} from '@wordpress/block-editor';
+import { Button, Flex, FlexItem } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { SVG, Path } from '@wordpress/primitives';
+
+import './editor.pcss';
+
+export default function Edit( { clientId, attributes, setAttributes } ) {
+	const blockProps = useBlockProps();
+	const dispatch = useDispatch( 'core/block-editor' );
+	const { removeBlocks } = useDispatch( 'core/block-editor' );
+	const select = useSelect( 'core/block-editor' );
+	const innerBlocks = select.getBlocks( clientId );
+	const { currentActiveTabInstanceId, tabs } = attributes;
+
+	// Update the current active tab to the client Id of the first tab
+	useEffect( () => {
+		if ( innerBlocks.length === 0 || currentActiveTabInstanceId !== '' ) {
```

Comment:
We're checking for `currentActiveTabInstanceId` here so that this is only running once when the Tabs block is added to a page. Not checking for this means that this runs on every re-render, and will set the active tab to the first tab on every re-render, which we obviously don't want when another tab has been selected.

To help this be understood a bit better I added an additional comment here.